src/hdmf/backends/hdf5/h5tools.py

```diff
@@ -699,7 +702,7 @@ def __read_dataset(self, h5obj, name=None):
             if h5obj.dtype.kind == 'O' and len(h5obj) > 0:
                 elem1 = h5obj[0]
                 if isinstance(elem1, (str, bytes)):
-                    d = h5obj
+                    d = self._check_str_dtype(h5obj)
```

Comment:
@ajtritt This only covers 1D string datasets. Multi-dim string datasets stay as H5Datasets that return as bytes objects. Can we cover multi-dim string datasets too?

MWE:

```python
from hdmf.backends.hdf5.h5tools import HDF5IO
from hdmf.build import DatasetBuilder
import os
import numpy as np

a = np.array([['a', 'bb'], ['ccc', 'dddd']])
attr = 'foobar'
path = 'test_h5py3.h5'
try:
    with HDF5IO(path, mode='a') as io:
        io.write_dataset(io._file, DatasetBuilder('test_dataset', a, attributes={'test_attr': attr}, dtype='text'))
        io.close()

    with HDF5IO(path, 'r') as io:
        bldr = io.read_builder()
        print(str(bldr['test_dataset'].data))
        print(bldr['test_dataset'].data[:])
        print(type(bldr['test_dataset'].data[0]))
finally:
    os.remove(path)
```

Reply:
If not, we should note this clearly in the release notes.

Reply:
How about now?

Reply:
Looks good. Thanks